sysvar/src/lib.rs

```diff
@@ -161,6 +161,27 @@ pub trait SysvarSerialize:
 /// Implements the [`Sysvar::get`] method for both SBF and host targets.
 #[macro_export]
 macro_rules! impl_sysvar_get {
+    () => {
+        fn get() -> Result<Self, $crate::__private::ProgramError> {
+            // Allocate uninitialized memory for the sysvar struct
+            let mut uninit = core::mem::MaybeUninit::<Self>::uninit();
+            let size = core::mem::size_of::<Self>() as u64;
+            // Safety: we build a mutable slice pointing to the uninitialized
+            // buffer.  The `get_sysvar` syscall will fill exactly `size`
+            // bytes, after which the buffer is fully initialised.
+            let dst = unsafe {
+                core::slice::from_raw_parts_mut(uninit.as_mut_ptr() as *mut u8, size as usize)
+            };
+            // Attempt to load the sysvar data.
+            $crate::get_sysvar(dst, &id(), 0, size)?;
```

Comment:
You probably don't need to create a slice, you can pass the mutable pointer from the `MaybeUninit`.
```suggestion
            $crate::get_sysvar(uninit..as_mut_ptr() as *mut _ as *mut u8, &id(), 0, size)?;
```